api/libproject/credentials.rb

```diff
@@ -0,0 +1,18 @@
+require_relative "../../libproject/utils/common"
+
+class Credentials
```

Comment:
It looks like these are per-environment constants? Calling this "Credentials" confused me, since I think this module actually does not have any responsibility for authorization / fetching creds; it's just storing constants, right?

Another pattern I've used is an `Environment` struct that gets filled out based on which environment is active, so you'd do something like `get_environment(:dev).sa_file_name`. That would also localize all the references to the `:dev`, `:prod`, etc keywords -- only the `get_environment` function would have to know about / validate them, and everyone else could just access fields of the struct.